### ISSUE TITLE
Cart add() function doesn't follow minimum requirements

### ISSUE THREAD

**Issue by thekrotek** [opencart/opencart#2914]
When you add to cart from module, add() function in cart's controller doesn't follow minimum requirements for product. Here's the code:
```php
if (isset($this->request->post['quantity'])) {
	$quantity = (int)$this->request->post['quantity'];
} else {
	$quantity = $product_info['minimum'] ? $product_info['minimum'] : 1;
}
```
It check for "quantity" variable in post array, but it always equals 1, when JS add() function from common.js is used. Fix is pretty simple:
```php
if (isset($this->request->post['quantity'])) {
	if ($product_info['minimum'] && ($this->request->post['quantity']  < $product_info['minimum'])) {
		$quantity = $product_info['minimum'];
	} else {
		$quantity = (int)$this->request->post['quantity'];
	}
} else {
	$quantity = $product_info['minimum'] ? $product_info['minimum'] : 1;
}
```

**Comment by Golling** [comment]
Well done !

I'm working on this file and I didn't see this miss. ^^

**Comment by stockmaster** [comment]
You could sugest the change in a Pull request TheKroteck, might be quickly implemented on the next RC build

**Comment by GomDani** [comment]
I change to
```
if (isset($this->request->post['quantity']) && ((int)$this->request->post['quantity'] >= $product_info['minimum'])) {
    $quantity = (int)$this->request->post['quantity'];
} else {
    $quantity = $product_info['minimum'] ? $product_info['minimum'] : 1;
}
```

**Comment by thekrotek** [comment]
@stockmaster Github confuses me A LOT. I honestly don't understand pull requests and markdown language instead of normal WYSIWYG editor. Sorry for posting this as an issue.

@GomDani Yeah, it's a shorter version.

**Comment by danielkerr** [comment]
pull reqaut please

**Comment by thekrotek** [comment]
In your mind.

**Comment by osworx** [comment]
@stockmaster:
1. you do not have to write in any markup, write it in plain text
2. to create a `pull request`, simple follow these steps:

A. locate in this repository (OpenCart) to the file you want to have your changes
B. see below image to create a `fork`
C. now edit this file in YOUR own repo
D. when ready, add a note everybody understands to your changes (bottom of page) and hit the button `Propose file change`

That's all.
![forking_a_file](https://cloud.githubusercontent.com/assets/1905204/7231888/90ece40c-e778-11e4-974f-a9aa1cf89eab.png)

Guess this is understandable by everyone - and the change for this request to get accepted and added to the core is much higher this way, instead of only writing a text message.

**Comment by stockmaster** [comment]
Admittedly Github is not the easiest thing to work with or get ones mind around it, but those instructions seem very clear! thank you for that
Also opensource is a 2 way street, give 1 take 1, at least that's the way i see it.
I had one of my pull requests added YAY! :D

**Comment by stockmaster** [comment]
@osworx not sure if you could explain this as well, but suppose i wanted to pull your Pull request to my fork, is there any 3 easy steps to do it or - as i read in most places - has to be done with command lines? Thanks in advance

**Comment by osworx** [comment]
Puhh .. I am not the GitHub expert .. do not know this.
Maybe someone else can answer?

**Comment by stockmaster** [comment]
What i been doing is, Delete fork -> get new fork from Opencart/opencart master once commits are added. but i am sure thats a stupid way to do it! lol

**Comment by stockmaster** [comment]
Mom i did it! No Hands!
with the below instructions

"I assume, you should have already a fork (MyFork) of the common repo (BaseRepo) which has the pending pull request from a fork (OtherFork) you are interested in.

Navigate to the fork (OtherFork) which has initiated the pull request which you like to get into your fork (MyFork)
Go to the pull requests page of OtherFork
Click new pull request
The pending pull request(s) should be offered. Select on the left side as the base fork your fork (MyFork) (IMPORTANT)
Now the option of View pull request should change to Create pull request. Click this.
Now you should have a pending pull request in your fork (MyFork), which you can simply accept.
"

**Comment by osworx** [comment]
Thx ;-)

**Comment by thekrotek** [comment]
@osworx, thnx for the tutorial, pal, but I'll skip creating any PRs here. This repo's owner clearly has no idea, how to provide support and encourage people to report issues. I've created an issue for PayPal currencies list here:

https://github.com/opencart/opencart/issues/2932

And the only reply I got was "in your mind". Guy intermediately closed the ticked with absolutely ignorant response. And this is not the first time he reacts like teenager, not like a pro. So let him fix everything by himself. I'll be fixing for my own OC installation and other people, well... they have to suck, what they got "out of the box".
